google/cloud/storage/client.cc

```diff
@@ -268,6 +268,36 @@ credentials, as described in:
   return std::move(os).str();
 }
 
+StatusOr<PolicyDocumentResult> Client::SignPolicyDocument(
+    internal::PolicyDocumentRequest const& request) {
+  auto base_credentials = raw_client()->client_options().credentials();
+  auto credentials = dynamic_cast<oauth2::ServiceAccountCredentials<>*>(
+      base_credentials.get());
+
+  if (credentials == nullptr) {
+    return Status(
+        StatusCode::kInvalidArgument,
+        R"""(The current credentials cannot be used to sign policy documents.
+Please configure your google::cloud::storage::Client to use service account
+credentials, as described in:
+https://cloud.google.com/storage/docs/authentication
+)""");
+  }
+
+  auto base64_policy = internal::Base64Encode(request.StringToSign());
+
+  auto result = credentials->SignString(base64_policy);
+  if (!result.first.ok()) {
```

Comment:
I copied this line from this example:

https://github.com/googleapis/google-cloud-cpp/blob/63fcd93928a6aaab1bf045bccb610938ba246b21/google/cloud/storage/client.cc#L217

But I am unable to see in what scenario this would never equal `ok()`. To add to that, the line is red in the code coverage report.

Reply:
I commented in https://github.com/googleapis/google-cloud-cpp/issues/1811 that `SignString` doesn't return `StatusOr`. But I attempted to fix it myself and realized that the call chain is actually pretty large, so I gave up.